code/modules/projectiles/ammunition.dm

```diff
@@ -315,6 +319,40 @@ Turn() or Shift() as there is virtually no overhead. ~N
 	icon_state = "base"
 	w_class = SIZE_HUGE
 	var/empty = FALSE
+	var/can_explode = FALSE
+	var/burning = FALSE
+	var/limit_per_tile = 1	//how many you can deploy per tile
+
+//---------------------GENERAL PROCS
+
+/obj/item/ammo_box/Destroy()
+	SetLuminosity(0)
+	. = ..()
+
+//---------------------FIRE HANDLING PROCS
+/obj/item/ammo_box/flamer_fire_act(var/severity, var/datum/cause_data/flame_cause_data)
+	if(burning)
+		return
+	burning = TRUE
+
+	SetLuminosity(3)
+	apply_fire_overlay()
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, src), 5 SECONDS)
+
+/obj/item/ammo_box/proc/apply_fire_overlay(var/will_explode = FALSE)
```

Comment:
maybe this should change colour with the colour of the fire like how burning mobs do?